smart-contract/assembly/contracts/internals/fileInit.ts

```diff
@@ -0,0 +1,57 @@
+import { sha256, Storage } from '@massalabs/massa-as-sdk';
+import { _pushFileLocation } from './location';
+import { fileMetadataLocationKey } from './storageKeys/metadataKeys';
+import { Metadata } from '../serializable/Metadata';
+import { _editFileMetadata } from './metadata';
+import {
+  _getTotalChunk,
+  _removeChunksRange,
+  _setFileChunkCount,
+} from './chunks';
+import { stringToBytes } from '@massalabs/as-types';
+
+/**
+ * Initializes a file in the smart contract storage.
+ * @param location - The location of the file.
+ * @param hashLocation - The hash of the file location.
+ * @param totalChunk - The total number of chunks for the file.
+ * @param metadata - An array of Metadata objects for the file.
+ * @throws If the total chunk count is 0 or less.
+ */
+export function _fileInit(
```

Comment:
according to the spec here: https://forum.massa.community/t/community-feedback-proposal-for-improving-website-storage-on-deweb/82/15#p-319-smart-contract-code-2 this function starts by deleting the file completely in order to remove any unexpected garbage (eg. metadata and so on). Any reason to depart from that ?